### PR TITLE
Check for (and send) empty body on log out post.

### LINKED ISSUE #132: Logout not Working (BlazorWebAssemblyStandaloneWithIdentity)
**Describe the issue**
After register a new user and login the logout is not working. The screen is flickering and the user is not logged out.

**To Reproduce**
1. Create a new project map with regarding projects (Backend and BlazorWasmAuth)
2. Set both project as start projects
3. Register a new user
4. Login for new user
5. Try to logout

**Expected behavior**
User should become logged out

**.NET Version**
8.0.100

---
#### Issue Details

⚠ *Do not edit this section. It is required for issue processing.*

* Content Source: [dotnet/blazor-samples](https://github.com/dotnet/blazor-samples)
* GitHub Login: @guardrex

### PR DESCRIPTION
**Pull request by JeremyLikness**
[EDIT by guardrex to add another 'fixes']

Fixes issue `dotnet/AspNetCore.Docs` [#31205](https://github.com/dotnet/AspNetCore.Docs/issues/31205)
Fixes #132

### CODE REVIEW

File: 8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/Program.cs [hunk 1]
```diff
@@ -42,13 +43,21 @@
...
-// NOTE: This logout code will be updated shortly.
-//       https://github.com/dotnet/blazor-samples/issues/132
-app.MapPost("/Logout", async (ClaimsPrincipal user, SignInManager<AppUser> signInManager) =>
+// The request checks for an empty body to prevent CSRF attacks. By requiring something
+// in the body, the request must be made from JavaScript, which is the only way to
+// access the cookie. It can't be accessed by a form-based post.
+// This prevents a malicious site from logging the user out.
+// Furthermore, the endpoint is protected by authorization to prevent anonymous access.
+// The client simply needs to pass an empty object {} in the body of the request.
+app.MapPost("/Logout", async (SignInManager<AppUser> signInManager, [FromBody]object empty) =>
```
Comment (halter73) [inline 8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/Program.cs:52]: @blowdart @GrabYourPitchforks Does this look reasonable to you?

**Review by halter73 — APPROVED** [review]

**Comment by halter73** [comment]
Can you add write access for me too?

**Comment by guardrex** [comment]
Yes, @halter73 ... and I messaged Artak about adding you guys to the Blazor engineering team, which will take care of it, too. For now, I'll place a manual entry in the repo for you.

File: 8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/Program.cs (hunk 1 above)
Comment (guardrex) [reply 8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/Program.cs:52]: @halter73 ... Looks like they didn't receive the ping. I don't have Levi's email. Do you want to ping them offline to look?

Comment (halter73) [reply 8.0/BlazorWebAssemblyStandaloneWithIdentity/Backend/Program.cs:52]:
```suggestion
app.MapPost("/Logout", async (SignInManager<AppUser> signInManager, [FromBody] object empty) =>
```

**Pull request merged**